### PR TITLE
ICU-21957 Fix docker for creating ICU4C distribution results

### PR DESCRIPTION
**Pull request by sven-oly**

### CODE REVIEW

**Comment by sven-oly** [comment]
I'm not sure that all of these changes are necessary (or sufficient).

File: sort-out-dist.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ do
     #echo "# ${file}"
     base=$(basename $file)
     # rver is '66-1'
-    rver=$(echo $file | grep "^icu-rrelease-[0-9][0-9]-[0-9]" | cut -d- -f3-4)
+    rver=$(echo $file | grep "^icu-rrelease-[0-9][0-9]-[0-9rc]" | cut -d- -f3-4)
```
Comment (markusicu) [inline sort-out-dist.sh:28]: Possibly a little tighter? Just guessing.
```suggestion
    rver=$(echo $file | grep "^icu-rrelease-[0-9][0-9]-[0-9](rc)?" | cut -d- -f3-4)
```

**Review by srl295 — CHANGES_REQUESTED** [review]
generally LGTM but agree the rc should be fixed

File: sort-out-dist.sh (hunk 1 above)
Comment (sven-oly) [reply sort-out-dist.sh:28]: No, the suggestion doesn't work with the file names that are created. Leaving as is!

**Comment by sven-oly** [comment]
FYI, this is now how the sort runs:
$ sudo ./sort-out-dist.sh 
'icu4c-72.rc/icu4c-72_rc-Fedora_Linux36-x64.tgz' => 'icu-rrelease-72-rc-x86_64-pc-linux-gnu-Fedora_Linux-36.tgz'
'icu4c-72.rc/icu4c-72_rc-Ubuntu22.04-x64.tgz' => 'icu-rrelease-72-rc-x86_64-pc-linux-gnu-Ubuntu-22.04-OLD.tgz'
'icu4c-72.rc/SOURCEDOC-icu4c-72_rc-SOURCEDOC.tgz' => 'icu-rrelease-72-rc-x86_64-pc-linux-gnu-Ubuntu-22.04-sdoc.tgz'
'icu4c-72.rc/icu4c-72_rc-docs.zip' => 'icu-rrelease-72-rc-x86_64-pc-linux-gnu-Ubuntu-22.04-src.d/icu4c-72_1-docs.zip'
## Warn: could not classify icu4c-72_1-release-72-rc-data.zip
'icu4c-72.rc/icu4c-72_rc-data-bin-b.zip' => 'icu-rrelease-72-rc-x86_64-pc-linux-gnu-Ubuntu-22.04-src.d/icu4c-72_1-data-bin-b.zip'
'icu4c-72.rc/icu4c-72_rc-data.zip' => 'icu-rrelease-72-rc-x86_64-pc-linux-gnu-Ubuntu-22.04-src.d/icu4c-72_1-data.zip'
'icu4c-72.rc/icu4c-72_rc-data-bin-l.zip' => 'icu-rrelease-72-rc-x86_64-pc-linux-gnu-Ubuntu-22.04-src.d/icu4c-72_1-data-bin-l.zip'
## Warn: could not classify icu4c-72_1-release-72-rc-src.zip
'icu4c-72.rc/icu4c-72_rc-src.tgz' => 'icu-rrelease-72-rc-x86_64-pc-linux-gnu-Ubuntu-22.04-src.d/icu4c-72_1-src.tgz'
## Warn: could not classify icu4c-72_1-release-72-rc-docs.zip
'icu4c-72.rc/icu4c-72_rc-src.zip' => 'icu-rrelease-72-rc-x86_64-pc-linux-gnu-Ubuntu-22.04-src.d/icu4c-72_1-src.zip'
## Warn: could not classify icu4c-72_1-release-72-rc-src.tgz
# exists: icu4c-72.rc/icu4c-72_rc-Ubuntu22.04-x64.tgz (not copying icu-rrelease-72-rc-x86_64-pc-linux-gnu-Ubuntu-22.04.tgz)


Giving these as the output of the sort:
$ ls -l dist/icu4c-72.rc
total 142440
-rw-r--r-- 2 1000 loas-corp-daemons 12555122 Sep 26 14:43 icu4c-72_rc-data-bin-b.zip
-rw-r--r-- 2 1000 loas-corp-daemons 12255652 Sep 26 14:43 icu4c-72_rc-data-bin-l.zip
-rw-r--r-- 3 1000 loas-corp-daemons 19608668 Sep 26 14:42 icu4c-72_rc-data.zip
-rw-r--r-- 3 1000 loas-corp-daemons  8395041 Sep 26 14:43 icu4c-72_rc-docs.zip
-rw-r--r-- 2 1000 loas-corp-daemons 16165441 Sep 26 17:37 icu4c-72_rc-Fedora_Linux36-x64.tgz
-rw-r--r-- 3 1000 loas-corp-daemons 26303390 Sep 26 14:43 icu4c-72_rc-src.tgz
-rw-r--r-- 3 1000 loas-corp-daemons 27616189 Sep 26 14:43 icu4c-72_rc-src.zip
-rw-r--r-- 2 1000 loas-corp-daemons 16205698 Sep 26 14:45 icu4c-72_rc-Ubuntu22.04-x64.tgz
-rw-r--r-- 2 1000 loas-corp-daemons  6728981 Sep 26 14:43 SOURCEDOC-icu4c-72_rc-SOURCEDOC.tgz

These files are named with the _rc version, ready for adding to the binary release.

**Pull request closed without merging**